core/src/main/java/com/orientechnologies/orient/core/storage/fs/OFileClassic.java

```diff
@@ -136,16 +133,15 @@ public void read(long offset, byte[] iData, int iLength, int iArrayOffset) throw
   }
 
   @Override
-  public void read(long offset, ByteBuffer buffer) throws IOException {
+  public void read(long offset, ByteBuffer buffer, boolean throwOnEof) throws IOException {
     int attempts = 0;
 
     while (true) {
       try {
         acquireReadLock();
         try {
           offset = checkRegions(offset, buffer.limit());
-
-          readByteBuffer(buffer, channel, offset);
+          readByteBuffer(buffer, channel, offset, throwOnEof);

```

Comment:
It was done in case OS automatically closed file descriptor. We just left it for the safety reasons.